Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp

```diff
@@ -277,7 +283,95 @@ AcceleratedSurfaceDMABuf::RenderTargetEGLImage::RenderTargetEGLImage(uint64_t su
 
     WebProcess::singleton().parentProcessConnection()->send(Messages::AcceleratedBackingStoreDMABuf::DidCreateBuffer(m_id, size, format, WTFMove(fds), WTFMove(offsets), WTFMove(strides), modifier, usage), surfaceID);
 }
+#endif // USE(GBM)
+
+#if OS(ANDROID)
+// TODO: There is another copy of this function in GLDisplay.cpp, remove one.
+
+static uint32_t
+fourccToAHardwareBufferFormat(uint32_t fourcc)
+{
+    switch (fourcc) {
+    case DRM_FORMAT_RGBX8888:
+        return AHARDWAREBUFFER_FORMAT_R8G8B8X8_UNORM;
+    case DRM_FORMAT_RGBA8888:
+        return AHARDWAREBUFFER_FORMAT_R8G8B8A8_UNORM;
+    case DRM_FORMAT_RGB565:
+        return AHARDWAREBUFFER_FORMAT_R5G6B5_UNORM;
+    case DRM_FORMAT_RGBA1010102:
+        return AHARDWAREBUFFER_FORMAT_R10G10B10A2_UNORM;
+    case DRM_FORMAT_RGB888:
+        return AHARDWAREBUFFER_FORMAT_R8G8B8_UNORM;
+    default:
+        RELEASE_ASSERT_NOT_REACHED();
+        return 0;
+    }
+}
+
+static uint64_t
+usageToAHardwareBufferUsage(DMABufRendererBufferFormat::Usage usage)
+{
+    switch (usage) {
+    case DMABufRendererBufferFormat::Usage::Rendering:
+    case DMABufRendererBufferFormat::Usage::Mapping:
+        return AHARDWAREBUFFER_USAGE_CPU_READ_RARELY | AHARDWAREBUFFER_USAGE_GPU_FRAMEBUFFER;
+    case DMABufRendererBufferFormat::Usage::Scanout:
+        return AHARDWAREBUFFER_USAGE_CPU_READ_RARELY | AHARDWAREBUFFER_USAGE_GPU_FRAMEBUFFER | AHARDWAREBUFFER_USAGE_FRONT_BUFFER;
+    }
+}
+
+std::unique_ptr<AcceleratedSurfaceDMABuf::RenderTarget> AcceleratedSurfaceDMABuf::RenderTargetEGLImage::create(uint64_t surfaceID, const WebCore::IntSize& size, const BufferFormat& dmabufFormat)
+{
+    if (!dmabufFormat.fourcc) {
+        WTFLogAlways("Failed to create GBM buffer of size %dx%d: no valid format found", size.width(), size.height());
+        return nullptr;
+    }
+
+    AHardwareBuffer_Desc description = { };
```

Comment:
s/ { } /

Reply:
I checked this, and as I was expecting we need either the trailing `= { }` or `{ }`  to tell the compiler to perform zero-initialization.